File: Arrays & Hashing/cpp/encode_and_decode.cpp
#include <vector>
using std::vector;
#include <string>
using std::string;

class Solution {
public:

    string encode(vector<string>& strs) {
        string together = ""; 
        for (string str : strs) {
            together += std::to_string(str.length()) + "#" + str; 
        }
        return together;
    }

    vector<string> decode(string s) {
        vector<string> lst; 
        string currLength = "";
        string currStr = "";
        for (int i = 0; i < s.length(); i++) {
            char currChar = s[i]; 
            if (currChar != '#') {
                currLength = currLength + currChar; 
            }
            else {
                int strLength = std::stoi(currLength);
                currStr = s.substr(++i, strLength); 
                lst.push_back(currStr);
                i += strLength - 1; 
                currLength = "";
            }
        }
        return lst; 
    }
};

File: Arrays & Hashing/java/encodeAndDecode.java
import java.util.*; 

public class encodeAndDecode {
    public String encode(List<String> strs) {
        String together = "";
        for (String str : strs) {
            together += str.length() + "#" + str; 
        }
        return together; 
    }

    public List<String> decode(String str) {
        List<String> lst = new ArrayList<>();
        String currLength = "";
        String currStr = "";
        for (int i = 0; i < str.length(); i++) {
            char currChar = str.charAt(i);
            if (currChar != '#') {
                currLength = currLength + currChar; 
            }
            else {
                int strLength = Integer.parseInt(currLength); 
                currStr = str.substring(++i, i+strLength); 
                lst.add(currStr);
                i += strLength - 1; 
                currLength = "";
            }
        }
        return lst; 
    }
}
